### PR TITLE
Fix Transpose ( issue #196 ), and add an example app.

### PR DESCRIPTION
**Pull request by ndpmurphy**
Transpose did not work.

## Test Plan

Examples/Transpose contains a demo that shows Views and Images transposed in various ways.
It shows that 2d transforms now work. Transforms with perspective seem to get clipped in some odd way and do not match the appearance when run on IOS. Also the transformed images have random background colours. So these are 2 things that could be fixed later.

### CODE REVIEW

File: ReactQt/runtime/src/reactitem.cpp [hunk 1]
```diff
@@ -393,6 +412,17 @@ double ReactItem::shadowRadius() const {
 
 void ReactItem::setShadowRadius(double shadowRadius) {}
 
+QVector<float> ReactItem::transform() const {
+    return QVector<float>{};
```
Comment (vkjr) [inline ReactQt/runtime/src/reactitem.cpp]: @ndpmurphy, please return here the same value that was passed to `setTransform()` function.
We can maintain `QVector<float> _transform` member, save value passed in `setTransform` to it and return it from `transform()`

File: Examples/Transform/index.js [hunk 2]
```diff
@@ -0,0 +1,167 @@
+
+// Display views and an image with various transforms applied
```
Comment (vkjr) [inline Examples/Transform/index.js]: @ndpmurphy, please, use the same naming for example js files as in another examples, `TransformApp.js` in this case.
Also example should be added to `Examples/CMakeLists.txt`

Comment (ndpmurphy) [reply Examples/Transform/index.js]: I have made this change, but can you tell me what is the standard way of running the Examples. I have been manually editing the main.qml.in file to set the source file and moduleName. This works, but I am sure the CMakefiles must provide a better way to launch an example from with QtCreator or from the commandline.

Comment (vkjr) [reply Examples/Transform/index.js]: @ndpmurphy, here is how you can setup "Run" section in QtCreator to launch examples using `qmlscene`:

screen shot 2018-03-20 at 8 59 39 am

Comment (ndpmurphy) [reply Examples/Transform/index.js]: But there is no Image.qml file in the Image directory. Was this supposed to be generated by CMakefile ?

Comment (vkjr) [reply Examples/Transform/index.js]: @ndpmurphy, this command invoked in `%{buildDir}`, which is build directory. You will get Image.qml there after build.

In `Examples/Image/CMakeLists.txt` you can find following code:
```
configure_file(
  ../Example.qml.in
  ${CMAKE_CURRENT_BINARY_DIR}/${EXAMPLE_NAME}.qml
)
```

It is responsible for generating `Image.qml`

Comment (ndpmurphy) [reply Examples/Transform/index.js]: That does not work for me. If I open the CMakefiel at the top level, then I can not  get it to build. I have been opening the CMakefile in the ReactQT directory and that works fine for most things, but it does not generate the qml files in Examples. When opening the project, is the CMAkefile at the top levelthe right file to open or is there another project file ?

Comment (ndpmurphy) [reply Examples/Transform/index.js]: Oh - ignore that I see my mistake now

Comment (vkjr) [reply Examples/Transform/index.js]: @ndpmurphy, were you able to run the example apps the way I mentioned above?
I usually open root CMakeLists.txt in QtCreator. Then you will be able to build both, ReactQt and Examples projects.

**Pull request merged**